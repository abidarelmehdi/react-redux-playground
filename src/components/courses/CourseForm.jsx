import React from "react";
import TextInput from "../core/Forms/TextInput";
import ListInput from "../core/Forms/ListInput";
import SubmitBtn from "../core/Forms/SubmitBtn";

export default function CourseForm({ course, authors, onChange, onSubmit }) {
  return (
    <form onSubmit={onSubmit} className="py-6 px-4 sm:px-6 lg:px-8">
      <div>
        <h3 className="text-2xl leading-6 font-medium text-gray-900">
          Course Form
        </h3>
        <p className="mt-1 max-w-2xl text-sm leading-5 text-gray-500">
          Update and create new courses.
        </p>
      </div>
      <div className="mt-6 sm:mt-5">
        <div className="sm:grid sm:grid-cols-3 sm:gap-4 sm:items-start sm:border-t sm:border-gray-200 sm:pt-5">
          <TextInput
            id="title"
            label="Title"
            name="title"
            value={course.title}
            onChange={onChange}
          />
        </div>

        <div className="mt-6 sm:mt-5 sm:grid sm:grid-cols-3 sm:gap-4 sm:items-start sm:border-t sm:border-gray-200 sm:pt-5">
          <TextInput
            label="Category"
            id="category"
            name="category"
            value={course.category}
            onChange={onChange}
          />
        </div>

        <div className="mt-6 sm:mt-5 sm:grid sm:grid-cols-3 sm:gap-4 sm:items-start sm:border-t sm:border-gray-200 sm:pt-5">
          <ListInput
            id="authorId"
            label="Author"
            name="authorId"
            value={course.authorId}
            objectList={authors}
            onChange={onChange}
          />
        </div>
        <div className="mt-6 sm:mt-5 flex justify-end sm:border-t sm:border-gray-200 sm:pt-5">
          <SubmitBtn label="submit" />
        </div>
      </div>
    </form>
  );
}
