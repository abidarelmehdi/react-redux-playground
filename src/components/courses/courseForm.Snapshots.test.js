import React from "react";
import CourseForm from "./CourseForm";
import renderer from "react-test-renderer";
import { courses, authors } from "../../../tools/mockData";

it("replace save button with saving state when form is saving", () => {
  const tree = renderer.create(
    <CourseForm
      course={courses[0]}
      authors={authors}
      onChange={jest.fn()}
      onSubmit={jest.fn()}
      saving
    />
  );

  expect(tree).toMatchSnapshot();
});

it("load save button instead of saving state when form is loaded", () => {
  const tree = renderer.create(
    <CourseForm
      course={courses[0]}
      authors={authors}
      onChange={jest.fn()}
      onSubmit={jest.fn()}
      saving={false}
    />
  );

  expect(tree).toMatchSnapshot();
});
