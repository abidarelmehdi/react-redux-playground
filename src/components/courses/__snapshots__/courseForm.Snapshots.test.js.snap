// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`load save button instead of saving state when form is loaded 1`] = `
<form
  className="py-6 px-4 sm:px-6 lg:px-8"
  onSubmit={[MockFunction]}
>
  <div>
    <h3
      className="text-2xl leading-6 font-medium text-gray-900"
    >
      Course Form
    </h3>
    <p
      className="mt-1 max-w-2xl text-sm leading-5 text-gray-500"
    >
      Update and create new courses.
    </p>
  </div>
  <div
    className="mt-6 sm:mt-5"
  >
    <div
      className="sm:grid sm:grid-cols-3 sm:gap-4 sm:items-start sm:border-t sm:border-gray-200 sm:pt-5"
    >
      <label
        className="block text-sm font-medium leading-5 text-gray-700 sm:mt-px sm:pt-2"
        htmlFor="title"
      >
        Title
      </label>
      <div
        className="mt-1 sm:mt-0 col-span-2"
      >
        <div
          className="w-full rounded-md shadow-sm"
        >
          <input
            className="form-input block w-full transition duration-150 ease-in-out sm:text-sm sm:leading-5"
            id="title"
            name="title"
            onChange={[MockFunction]}
            type="text"
            value="Securing React Apps with Auth0"
          />
        </div>
      </div>
    </div>
    <div
      className="mt-6 sm:mt-5 sm:grid sm:grid-cols-3 sm:gap-4 sm:items-start sm:border-t sm:border-gray-200 sm:pt-5"
    >
      <label
        className="block text-sm font-medium leading-5 text-gray-700 sm:mt-px sm:pt-2"
        htmlFor="category"
      >
        Category
      </label>
      <div
        className="mt-1 sm:mt-0 col-span-2"
      >
        <div
          className="w-full rounded-md shadow-sm"
        >
          <input
            className="form-input block w-full transition duration-150 ease-in-out sm:text-sm sm:leading-5"
            id="category"
            name="category"
            onChange={[MockFunction]}
            type="text"
            value="JavaScript"
          />
        </div>
      </div>
    </div>
    <div
      className="mt-6 sm:mt-5 sm:grid sm:grid-cols-3 sm:gap-4 sm:items-start sm:border-t sm:border-gray-200 sm:pt-5"
    >
      <label
        className="block text-sm font-medium leading-5 text-gray-700 sm:mt-px sm:pt-2"
        htmlFor="authorId"
      >
        Author
      </label>
      <div
        className="mt-1 sm:mt-0 sm:col-span-2"
      >
        <div
          className="w-full rounded-md shadow-sm"
        >
          <select
            className="block w-full form-select transition duration-150 ease-in-out sm:text-sm sm:leading-5"
            id="authorId"
            name="authorId"
            onChange={[MockFunction]}
            value={1}
          >
            <option
              value=""
            >
              --------
            </option>
            <option
              value={1}
            >
              Cory House
            </option>
            <option
              value={2}
            >
              Scott Allen
            </option>
            <option
              value={3}
            >
              Dan Wahlin
            </option>
          </select>
        </div>
      </div>
    </div>
    <div
      className="mt-6 sm:mt-5 flex justify-end sm:border-t sm:border-gray-200 sm:pt-5"
    >
      <button
        className="px-4 py-2 border border-transparent text-base leading-6 font-medium rounded-md text-indigo-700 bg-indigo-100 hover:bg-indigo-50 focus:outline-none focus:border-indigo-300 focus:shadow-outline-indigo active:bg-indigo-200 transition ease-in-out duration-150"
        type="submit"
      >
        submit
      </button>
    </div>
  </div>
</form>
`;

exports[`replace save button with saving state when form is saving 1`] = `
<form
  className="py-6 px-4 sm:px-6 lg:px-8"
  onSubmit={[MockFunction]}
>
  <div>
    <h3
      className="text-2xl leading-6 font-medium text-gray-900"
    >
      Course Form
    </h3>
    <p
      className="mt-1 max-w-2xl text-sm leading-5 text-gray-500"
    >
      Update and create new courses.
    </p>
  </div>
  <div
    className="mt-6 sm:mt-5"
  >
    <div
      className="sm:grid sm:grid-cols-3 sm:gap-4 sm:items-start sm:border-t sm:border-gray-200 sm:pt-5"
    >
      <label
        className="block text-sm font-medium leading-5 text-gray-700 sm:mt-px sm:pt-2"
        htmlFor="title"
      >
        Title
      </label>
      <div
        className="mt-1 sm:mt-0 col-span-2"
      >
        <div
          className="w-full rounded-md shadow-sm"
        >
          <input
            className="form-input block w-full transition duration-150 ease-in-out sm:text-sm sm:leading-5"
            id="title"
            name="title"
            onChange={[MockFunction]}
            type="text"
            value="Securing React Apps with Auth0"
          />
        </div>
      </div>
    </div>
    <div
      className="mt-6 sm:mt-5 sm:grid sm:grid-cols-3 sm:gap-4 sm:items-start sm:border-t sm:border-gray-200 sm:pt-5"
    >
      <label
        className="block text-sm font-medium leading-5 text-gray-700 sm:mt-px sm:pt-2"
        htmlFor="category"
      >
        Category
      </label>
      <div
        className="mt-1 sm:mt-0 col-span-2"
      >
        <div
          className="w-full rounded-md shadow-sm"
        >
          <input
            className="form-input block w-full transition duration-150 ease-in-out sm:text-sm sm:leading-5"
            id="category"
            name="category"
            onChange={[MockFunction]}
            type="text"
            value="JavaScript"
          />
        </div>
      </div>
    </div>
    <div
      className="mt-6 sm:mt-5 sm:grid sm:grid-cols-3 sm:gap-4 sm:items-start sm:border-t sm:border-gray-200 sm:pt-5"
    >
      <label
        className="block text-sm font-medium leading-5 text-gray-700 sm:mt-px sm:pt-2"
        htmlFor="authorId"
      >
        Author
      </label>
      <div
        className="mt-1 sm:mt-0 sm:col-span-2"
      >
        <div
          className="w-full rounded-md shadow-sm"
        >
          <select
            className="block w-full form-select transition duration-150 ease-in-out sm:text-sm sm:leading-5"
            id="authorId"
            name="authorId"
            onChange={[MockFunction]}
            value={1}
          >
            <option
              value=""
            >
              --------
            </option>
            <option
              value={1}
            >
              Cory House
            </option>
            <option
              value={2}
            >
              Scott Allen
            </option>
            <option
              value={3}
            >
              Dan Wahlin
            </option>
          </select>
        </div>
      </div>
    </div>
    <div
      className="mt-6 sm:mt-5 flex justify-end sm:border-t sm:border-gray-200 sm:pt-5"
    >
      <div
        className="stage__Wrapper-sc-161krao-0 iPJbLy"
        size={40}
      >
        <div
          className="stage__Ball-sc-161krao-1 VHhMl"
          color="#252f3f"
          size={40}
          x={0}
          y={16}
        />
        <div
          className="stage__Ball-sc-161krao-1 dEEzDQ"
          color="#252f3f"
          size={40}
          x={16}
          y={16}
        />
        <div
          className="stage__Ball-sc-161krao-1 jbVSnv"
          color="#252f3f"
          size={40}
          x={32}
          y={16}
        />
      </div>
    </div>
  </div>
</form>
`;
